src/pages/Signup/Signup.jsx

```diff
@@ -1,7 +1,87 @@
-import React from 'react';
+import React, { useState } from 'react';
+import './Signup.scss';
 
-function Signup(props) {
-  return <div>여긴 회원가입이다</div>;
-}
+const Signup = () => {
+  const [userName, setUserName] = useState('');
+  const [userId, setUserId] = useState('');
+  const [userPw, setUserPw] = useState('');
+  const [userPwCheck, setUserPwCheck] = useState('');
+  const [buttonOn, setButtonOn] = useState(false);
+
+  const handleUserName = event => {
+    setUserName(event.target.value);
+    console.log(setUserName);
+  };
+
+  const handleUserId = event => {
+    setUserId(event.target.value);
+    console.log(setUserId);
+  };
+
+  const handleUserPw = event => {
+    setUserPw(event.target.value);
+    console.log(setUserPw);
+  };
+
+  const handleUserPwCheck = event => {
+    setUserPwCheck(event.target.value);
+    console.log(setUserPwCheck);
+  };
+
+  const isPassedId = () => {
+    return userId.includes('a-z0-9') && 16 > userId.length > 4
+      ? setButtonOn(true)
+      : setButtonOn(false);
+  };
```

Comment:
```suggestion
  const isPassedId = userId.includes('a-z0-9') && 16 > userId.length > 4;
```
buttonOn이라는 상태값으로 관리하기 보다는 위와 같이 변수로 사용해보세요!
상태값이 업데이트 될 때마다 재렌더링이 된다는 특징을 생각해보세요!